src/heliaFetch.ts

```diff
@@ -231,26 +235,21 @@ export class HeliaFetch {
    */
   private async getDirectoryResponse (...[cid, options]: Parameters<UnixFS['cat']>): Promise<AsyncIterable<Uint8Array>> {
     this.log('Getting directory response:', { cid, options })
-    const rootFile = await pTryEach(this.rootFilePatterns.map(file => {
-      const directoryPath = options?.path ?? ''
-      return async (): Promise<{ name: string, cid: CID }> => {
-        try {
-          const path = this.sanitizeUrlPath(`${directoryPath}/${file}`)
-          this.log('Trying to get root file:', { file, directoryPath })
-          const stats = await this.fs.stat(cid, { path })
-          this.log('Got root file:', { file, directoryPath, stats })
-          return {
-            name: file,
-            cid: stats.cid
-          }
-        } catch (error) {
-          return Promise.reject(error)
-        }
+    let rootFile: UnixFSEntry | null = null
+    for await (const file of this.fs.ls(cid, { signal: options?.signal })) {
+      if (this.rootFilePatterns.includes(file.name)) {
+        this.log(`Found root file '${file.name}': `, file)
+        rootFile = file
+        break
+      } else {
+        this.log(`Skipping ${file.type} '${file.name}' in root CID because the filename is not in rootFilePatterns: ${this.rootFilePatterns}`)
       }
-    }))
+    }
+    if (rootFile == null) {
+      throw new Error('No root file found')
+    }
```

Comment:
switch to fs.ls iterator over files we *KNOW* exist in the directory, and that we're likely to have blocks for, instead of spamming network requests for content that *may* exist. (thanks @achingbrain )